src/rocksdb.rs

```diff
@@ -83,7 +83,8 @@ impl <'a> Iterator for SubDBIterator<'a> {
             let key = unsafe { slice::from_raw_parts(key_ptr, key_len as usize) };
             let val_ptr = unsafe { rocksdb_ffi::rocksdb_iter_value(native_iter, val_len_ptr) };
             let val = unsafe { slice::from_raw_parts(val_ptr, val_len as usize) };
-            Some((key,val))
+
+            Some((key.to_vec().into_boxed_slice(),val.to_vec().into_boxed_slice()))
```

Comment:
This requires a copy--once you call iter_next, the previous pointer is corrupt. That's the problem I ran into while writing integration tests.

The next thing I'd like to do is add a binary codec trait, and then make the RocksDB be parameterized on the actual type. That way, register-sized types could be iterated/retrieved as values, and reference types could do their own custom codec, so that the API is in terms of the actual application logic. I'll probably initially do support for [u8], String, and u64.